src/imperative/imperative_utils.h

```diff
@@ -595,7 +595,10 @@ inline bool CheckAndInferShape(nnvm::Graph* p_g, mxnet::ShapeVector&& shapes,
     *contain_unknown = false;
   }
   nnvm::Graph& g = *p_g;
-  if (g.attrs.count("shape")) {
+  if (use_inputs) {
```

Comment:
Please add the comments in the code

Reply:
This code is removed from previous commit by mistake. This is just to recover them back.